src/run/run.c

```diff
@@ -1703,20 +1703,20 @@ static void run_context_detach_bus(RunContext *c) {
         c->match_disconnected = sd_bus_slot_unref(c->match_disconnected);
 }
 
-static int pty_forward_handler(PTYForward *f, int rcode, void *userdata) {
+static int pty_forward_handler(sd_event *e, PTYForward *f, int rcode, void *userdata) {
```

Comment:
nah, the event loop is also available under `f->event` anyway, no need to add a new param here

Reply:
@poettering It's not, it's been unreffed by the time this handler is invoked.

Reply:
or in other words, add ptyfwd_get_event() that returns the event loop object

Reply:
> @poettering It's not, it's been unreffed by the time this handler is invoked.

hmm, really, that sounds wrong then?

Reply:
@poettering It's intentional, it's reffed in a local variable to keep it alive. Please see `pty_forward_done()`